colossalai/utils/memory_tracer/allocator.py

```diff
@@ -0,0 +1,67 @@
+import torch
+from colossalai.utils.commons.singleton_meta import SingletonMeta
+from colossalai.zero.sharded_param import ShardedTensor
+
+from typing import Union
+
+
+def col_tensor_mem_usage(t: Union[torch.Tensor, ShardedTensor]) -> int:
+    if isinstance(t, ShardedTensor):
+        target = t.payload
+    else:
+        target = t
+    unit_in_byte = 0
+    if target.dtype == torch.half:
+        unit_in_byte = 2
```

Comment:
I think we can use `target.element_size()` here.

Reply:
Good idea, DONE